crates/starknet-server/src/api/json_rpc/spec_reader/spec_schemas/mod.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+
+use self::all_of_schema::AllOf;
+use self::array_primitive::ArrayPrimitive;
+use self::boolean_primitive::BooleanPrimitive;
+use self::integer_primitive::IntegerPrimitive;
+use self::object_primitive::ObjectPrimitive;
+use self::one_of_schema::OneOf;
+use self::ref_schema::Reference;
+use self::string_primitive::StringPrimitive;
+
+pub(crate) mod all_of_schema;
+pub(crate) mod array_primitive;
+pub(crate) mod boolean_primitive;
+pub(crate) mod integer_primitive;
+pub(crate) mod object_primitive;
+pub(crate) mod one_of_schema;
+pub(crate) mod ref_schema;
+pub(crate) mod string_primitive;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Common {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub title: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "additionalProperties")]
+    pub additional_properties: Option<bool>,
+    #[serde(rename = "type")]
+    pub t: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(tag = "type", rename_all = "lowercase")]
+pub enum Primitive {
+    Array(ArrayPrimitive),
+    Boolean(BooleanPrimitive),
+    Integer(IntegerPrimitive),
+    Number(IntegerPrimitive),
```

Comment:
Why is `Number` needed?

Reply:
because there are places in the spec where the type of the field is marked with number and other where its marked with integer. Due to my assumption that both represent the same types and an effort to change all occurences from number to integer is error prone, decided to add additional variant.

Reply:
Ok, it seems that type number exists in RPC spec 0.5.0, but not anymore in 0.6.0-rc0. We can keep it like this for now and I'll resolve the conversation.